collector/system_collector.go

```diff
@@ -109,13 +208,23 @@ func (sc *systemCollector) collectClusterStatusMetrics() (systemStatusMetrics []
 }
 
 func (sc *systemCollector) collectClusterNodeMetrics() (systemStatusMetrics []systemStatusMetric) {
-	clusterNodesStatus, err := sc.systemClient.ReadClusterNodesAggregateStatus()
+	clusterNodes, err := sc.systemClient.ReadClusterNodesAggregateStatus()
 	if err != nil {
 		level.Error(sc.logger).Log("msg", "Unable to collect cluster nodes status")
 		return
 	}
 
-	for _, c := range clusterNodesStatus.ControllerCluster {
+	controllerStatusMetrics := sc.extractControllerStatusMetrics(clusterNodes)
+	systemStatusMetrics = append(systemStatusMetrics, controllerStatusMetrics...)
+
+	managementNodeMetrics := sc.extractManagementNodeMetrics(clusterNodes)
+	systemStatusMetrics = append(systemStatusMetrics, managementNodeMetrics...)
+
+	return
+}
+
+func (sc *systemCollector) extractControllerStatusMetrics(clusterNodes administration.ClustersAggregateInfo) (systemStatusMetrics []systemStatusMetric) {
```

Comment:
The signature looks the same with extractManagementNodeMetrics.
If it only needs ControllerCluster, can we just pass it instead?